lib/remove_duplicate_emails.rb

```diff
@@ -0,0 +1,36 @@
+class RemoveDuplicateEmails
+  def run(do_it: false)
+    # find all duplicate contact infos, older than X
+    all_dup_contact_infos = ContactInfo.where("created_at < ?", Date.parse("1/10/2020")).select(:type, :value).group(:type, :value).having('count(*) > 1')
+    users_to_remove = [], contactinfos_to_remove = []
+
+    all_dup_contact_infos.each do |dup_contact_info|
+      # of the duplicate contact infos value ... find the unverified duplicates
+      unverified_contactinfos = ContactInfo.where(value: dup_contact_info.value, verified: false)
+
+      unverified_contactinfos.each do |unverified_contactinfo|
+        if unverified_contactinfo.user.contact_infos.count == 1 && unverified_contactinfo.user.contact_infos.first.value == unverified_contactinfo.value
+          # this is a dup user with just this one unverified email
+          users_to_remove.append(unverified_contactinfo.user.id)
```

Comment:
It seems to me the contact_infos table is missing a foreign key constraint to auto-delete when you delete the user, so you have to add it to contactinfos_to_remove here too.

Reply:
@Dantemss isnt this that? 

```
  has_many :contact_infos, dependent: :destroy, inverse_of: :user
```

Reply:
Ah you're right. I thought you had used `delete_all` which wouldn't use the dependent, but it's in fact `destroy_all` which does.